tests/control/test_KinematicSimulationTrajectoryExecutor.cpp

```diff
@@ -155,14 +155,23 @@ TEST_F(KinematicSimulationTrajectoryExecutorTest, execute_TrajectoryIsAlreadyRun
 
   auto future = executor.execute(mTraj);
   
-  // Executor cannot not immediately execute the next one.
-  EXPECT_THROW(executor.execute(mTraj), std::runtime_error);
+  // Executor possibly not able to immediately execute the next one.
+  try
+  {
+    executor.execute(mTraj);
+  }
+  catch (const std::runtime_error& e)
+  {
+    EXPECT_EQ(std::string(e.what()), "Another trajectory in execution.");
+
+    // It could not have finished the trajectory execution yet.
+    EXPECT_NE(mSkeleton->getDof(0)->getPosition(), 1.0);
```

Comment:
Does this always hold? What if the other thread finished the execution of line 156 by the time we got here?